react/week4/src/components/InputForm.js

```diff
@@ -0,0 +1,23 @@
+import React, { useContext } from "react";
+import { userListContext } from "../App";
+
+const InputForm = () => {
+  const { input, setInput, setLoading } = useContext(userListContext);
+  return (
+    <>
+      <form>
+          <input
+            type="text"
+            value={input}
+            placeholder="User name"
+            onChange={event => {
+              setInput(event.target.value);

```

Comment:
I saw you started working on the `AbortController` thingy, but still want to mention that it's common practice to delay the network requests when interacting with keystrokes with the technique called `debouncing`. Please read [this article](https://css-tricks.com/debouncing-throttling-explained-examples/) about what is it, how it works and how it differs from the `throttling` technique.

Reply:
thanks I will look on it